pkg/virt-launcher/virtwrap/network/network.go

```diff
@@ -0,0 +1,344 @@
+	}
+	nic.Gateway = routes[0].Gw
+
+	// Get interface MAC address
+	mac, err := Handler.GetMacDetails(podInterface)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to get MAC for %s", podInterface)
+		return nil, err
+	}
+	nic.MAC = mac
+	return nicLink, nil
+}
+
+func preparePodNetworkInterfaces(nic *VIF, nicLink netlink.Link) error {
+	// Remove IP from POD interface
+	err := Handler.AddrDel(nicLink, &nic.IP)
+
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to delete link for interface: %s", podInterface)
+		return err
+	}
+
+	// Set interface link to down to change its MAC address
+	err = Handler.LinkSetDown(nicLink)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to bring link down for interface: %s", podInterface)
+		return err
+	}
+
+	_, err = Handler.ChangeMacAddr(podInterface)
+	if err != nil {
+		return err
+	}
+
+	err = Handler.LinkSetUp(nicLink)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to bring link up for interface: %s", podInterface)
+		return err
+	}
+
+	// Create a macvlan link
+	macvlan := &netlink.Macvlan{
+		LinkAttrs: netlink.LinkAttrs{
+			Name:        macVlanIfaceName,
+			ParentIndex: nicLink.Attrs().Index,
+		},
+		Mode: netlink.MACVLAN_MODE_BRIDGE,
+	}
+
+	//Create macvlan interface
+	if err := Handler.LinkAdd(macvlan); err != nil {
+		log.Log.Reason(err).Errorf("failed to create macvlan interface")
+		return err
+	}
+
+	//get macvlan link
+	macvlink, err := Handler.LinkByName(macVlanIfaceName)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to get a link for interface: %s", macVlanIfaceName)
+		return err
+	}
+	err = Handler.LinkSetUp(macvlink)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to bring link up for interface: %s", macVlanIfaceName)
+		return err
+	}
+
+	// set fake ip on macvlan interface
+	fakeaddr, err := Handler.ParseAddr(macVlanFakeIP)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to bring link up for interface: %s", macVlanIfaceName)
+		return err
+	}
+
+	if err := Handler.AddrAdd(macvlink, fakeaddr); err != nil {
+		log.Log.Reason(err).Errorf("failed to set macvlan IP")
+		return err
+	}
+
+	return nil
+}
+
+func plugNetworkDevice(vm *v1.VirtualMachine, domain *api.Domain, vif *VIF) error {
+
+	// get VIF config
+	ifconf, err := decorateInterfaceConfig(vif)
+	if err != nil {
+		log.Log.Reason(err).Error("failed to get VIF config.")
+		return err
+	}
+
+	//TODO:(vladikr) Currently we support only one interface per vm. Improve this once we'll start supporting more.
+	if len(domain.Spec.Devices.Interfaces) == 0 {
+		domain.Spec.Devices.Interfaces = append(domain.Spec.Devices.Interfaces, *ifconf)
+	}
+	for idx, _ := range domain.Spec.Devices.Interfaces {
+		domain.Spec.Devices.Interfaces[idx] = *ifconf
+	}
+
+	// Update vm status with interface details
+	ifaceStatus := decorateInterfaceStatus(vif)
+	vm.Status.Interfaces = append(vm.Status.Interfaces, *ifaceStatus)
```

Comment:
Setting the interface here won't result in the updated VM object getting persisted on teh cluster.

Reply:
> Setting the interface here won't result in the updated VM object getting persisted on teh cluster.

Yea, I did the change locally. I'll push soon